### PR TITLE
[NEXT-9118][NEXT-9230] Add handling for multiple pages in plugin

### LINKED ISSUE #30: NEXT-9230 - Sorting in the PluginManager destroys the manager
Description: 
If you sort the LIst of the plug-ins in the Plugin Manager (e.g. by "Status"), you can no longer call up the configuration of the plugin. That means: clicking on three dots (in the right column) opens a small menu, where the entry "Configuration" is missing.

Environment: Shopware 6, v6.2.0, Admin, Plugin Manager

Steps to reproduction: 
1. open Plugin Manager
2. check that at least one plugin offers a possibility to configure (click on three points at this plugin, check that "configure" is included).
3. sort the list of plugins by clicking on "Status".
4. click on three points again in the sorted list and on the previously checked plugin -> "Configure" is not included.

Expected result: 
After sorting the list you should be able to configure the plugin.

Actual result: 
After sorting the list, the configuration of the plugin is not possible.

(https://sbp-issuetracker-attachments.s3.eu-west-1.amazonaws.com/1592301421_5ee8976d864e3_1592301413_5ee8976516da8_Auswahl_081.png) [Auswahl_080.png](https://sbp-issuetracker-attachments.s3.eu-west-1.amazonaws.com/1592301421_5ee8976dbf0c1_1592301413_5ee897653fb9f_Auswahl_080.png)

### LINKED ISSUE #38: NEXT-9118 - Config options dissapear on page 2 (plugins)
If you have more than 25 plugins installed and go to page two. The options under the 3 dots with Description and Config disapears. 

If you change Items per page to 50, so all plugins are on the same page it works fine again.

### PR DESCRIPTION
**Pull request by jinnoflife**
### 1. What does this change do, exactly?
Resolves #30 & #38

### 2. Describe each step to reproduce the issue or behaviour.
Try to configure plugins on the second page

### 3. Proof
https://vimeo.com/user87685972/review/458992620/4a90ec8297

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js [hunk 1]
```diff
@@ -396,8 +404,6 @@ Component.register('sw-plugin-list', {
                         configAvailable: false
                     };
                 });
-            })).finally(() => {
-                this.isLoading = false;
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js:400]: Why did you remove this?

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig [hunk 2]
```diff
@@ -84,7 +85,7 @@
                                 {% endblock %}
 
                                 <template v-if="item.customFields">
-                                    {% block sw_plugin_list_grid_columns_actions_settings %}
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig:87]: You rename the block. This could lead to plugin breaks.

**Review by jleifeld — CHANGES_REQUESTED** [review]

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js (hunk 1 above)
Comment (jinnoflife) [reply src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js:400]: Was not needed from my pov

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig (hunk 2 above)
Comment (jinnoflife) [reply src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig]: Absolutely right. I reverted this change

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js [hunk 3]
```diff
@@ -58,8 +58,13 @@ Component.register('sw-plugin-list', {
         },
 
         pluginCriteria() {
+            if (this.listingCriteria) {
+                return this.listingCriteria;
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js:62]: what happen when the listingCriteria exists and the user changes the sortDirection or uses a search term?

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js [hunk 4]
```diff
@@ -355,10 +360,12 @@ Component.register('sw-plugin-list', {
...
         },
 
-        getList() {
+        getList(searchCriteria) {
             this.isLoading = true;
 
-            return State.dispatch('swPlugin/updatePluginList', this.searchData)
+            this.listingCriteria = searchCriteria;
+
+            State.dispatch('swPlugin/updatePluginList', this.searchData)
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js:368]: This would break the `getList` method. It returned a Promise before and now it is a void function. This can break plugins :)

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js [hunk 5]
```diff
@@ -378,13 +385,13 @@ Component.register('sw-plugin-list', {
         isConfigAvailableForPlugins() {
             this.isLoading = true;
 
-            return Promise.all(this.plugins.map((plugin) => {
+            this.plugins.map((plugin) => {
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/index.js:388]: You create a new variable and don´t return it. Whats the purpose of this? :) And also this could leak to potential breaks because the methods return previously a Promise and now is a void function

File: src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig [hunk 6]
```diff
@@ -43,7 +43,8 @@
                             :showPreviews="true"
                             :sortBy="sortBy"
                             :sortDirection="sortDirection"
-                            :compactMode="false">
+                            :compactMode="false"
+                            @doSearch="getList">
```
Comment (jleifeld) [inline src/Administration/Resources/app/administration/src/module/sw-plugin/view/sw-plugin-list/sw-plugin-list.html.twig:47]: The `sw-entiity-listing` should support pagination out of the box. I think it is better to find out the source of the problem then to fix the symptoms :)

**Review by jleifeld — CHANGES_REQUESTED** [review]

**Comment by shyim** [comment]
As this PR is inactive, I will close this here. Feel free to create an updated PR at shopware/platform
